### PR TITLE
Support Dark Mode in demo site

### PR DESCRIPTION
**Pull request by JiuqingSong**
- Support dark mode in demo site
- Add dark mode related Code object to demo site
- Add "getDarkColor" to API playground
- Do necessary refactoring in roosterjs code
- Create a new package roosterjs-color-util to do color calculation
- Add new EditorOption item getDarkColor to specify the color transform function

### CODE REVIEW

File: packages/roosterjs-color-utils/lib/utils/getDarkColor.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Get dark mode color for a given color
+ * @param color The color to calculate from
+ */
+export default function getDarkColor(color: string): string {
+    try {
+        const computedColor = Color(color || undefined);
+        const colorLab = computedColor.lab().array();
+        const newLValue = (100 - colorLab[0]) * ((100 - baseLValue) / 100) + baseLValue;
```
Comment (flyingbee2012) [inline packages/roosterjs-color-utils/lib/utils/getDarkColor.ts:13]: Not sure if this is the standard way to get the dark mode color?

File: demo/scripts/controls/MainPane.tsx [hunk 2]
```diff
@@ -134,19 +142,14 @@ class MainPane extends MainPaneBase {
...
-
         this.setState({
             isPopoutShown: true,
         });
 
         this.popoutRoot = win.document.getElementById(PopoutRoot);
 
         window.setTimeout(() => {
-            ReactDom.render(
-                <PopoutMainPane ref={this.popoutMainPane} content={this.content} />,
```
[the remark this answers is not in the document]
Comment (JiuqingSong) [reply demo/scripts/controls/MainPane.tsx:147]: Yes. There is a plugin called "Bridge" to help manage this kind of stuff.

File: demo/scripts/controls/editor/Editor.tsx [hunk 3]
```diff
@@ -10,7 +10,8 @@ import { CustomReplace as CustomReplacePlugin } from 'roosterjs-editor-plugins/l
 import { CutPasteListChain } from 'roosterjs-editor-plugins/lib/CutPasteListChain';
```
[the remark this answers is not in the document]
Comment (JiuqingSong) [reply demo/scripts/controls/editor/Editor.tsx:10]: There are some old style code that help management the lifecycle. To make dark mode work consistantly when switch back and force, I moved them out into Bridge plugin, so other code can be refactored into functional component.

For production code I agree we'd better separate it to be another PR, but given that it is the demo site code, I think we can keep it here.

File: packages/roosterjs-color-utils/lib/utils/getDarkColor.ts (hunk 1 above)
Comment (JiuqingSong) [reply packages/roosterjs-color-utils/lib/utils/getDarkColor.ts:13]: This is copied from OWA

**Review by flyingbee2012 — APPROVED** [review]

**Pull request merged**